packages/vite/src/migrations/update-15-3-4/set-mode-in-configuration.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  formatFiles,
+  readProjectConfiguration,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+
+export async function setModeInConfiguration(tree: Tree) {
+  forAllProjectsUsingViteAddMode(tree);
+  await formatFiles(tree);
+}
+
+export default setModeInConfiguration;
+
+function forAllProjectsUsingViteAddMode(tree: Tree): void {
+  forEachExecutorOptions(
+    tree,
+    '@nrwl/vite:build',
+    (_options, projectName, targetName) => {
```

Comment:
This callback also gets the `configuration` name. And each configuration setting. So you can use that to make the adjustments.